insights/tests/client/apps/test_malware_detection.py

```diff
@@ -0,0 +1,1287 @@
+        assert exc_info.value.code == constants.sig_kill_bad
+
+    def test_process_include_exclude_items_complex(self, conf, yara, rules, cmd):
+        # Test the process function with modified include and exclude files that will require more complex
+        # processing to generate the list of items to be scanned
+        # Because we are including items in /var/lib, we only need to list the contents of the /var/lib directory
+        # We don't need to list the contents of the /var directory
+        mdc = MalwareDetectionClient(None)
+        mdc.scan_fsobjects = ['/var/lib', '/var/log']
+        mdc.scan_exclude_list = ['/var/lib/systemd', '/var/lib/misc/', '/var/log/wtmp']
+
+        scan_dict = process_include_exclude_items(include_items=mdc.scan_fsobjects,
+                                                  exclude_items=mdc.scan_exclude_list)
+        assert list(scan_dict.keys()) == ['/var']
+        assert sorted(scan_dict['/var']['exclude']['items']) == ['/var/lib/misc', '/var/lib/systemd', '/var/log/wtmp']
+        # The exclude items shouldn't be in the include items
+        # Nor should other items that aren't in the explicitly included items
+        assert all([x not in scan_dict['/var']['include']
+                    for x in ['/var/lib/misc', '/var/lib/systemd', '/var/log/wtmp',
+                              '/var/cache', '/var/lib', '/var/log', '/var/tmp', '/tmp']])
+        # In 'include' will be items that are in the same directory as the excluded items, eg /var/log/lastlog
+        # but not the excluded items, eg /var/log/wtmp
+        assert all([x in scan_dict['/var']['include']
+                    for x in ['/var/lib/xkb', '/var/lib/dbus', '/var/log/btmp', '/var/log/lastlog']])
+
+        # Change the include directory to /var
+        # Now immediate child directories of /var will be in the include list, eg /var/cache and /var/tmp
+        # Because now we have to list the contents of the /var and /var/lib directories
+        mdc.scan_fsobjects.append('/var')
+        scan_dict = process_include_exclude_items(include_items=mdc.scan_fsobjects,
+                                                  exclude_items=mdc.scan_exclude_list)
+        assert list(scan_dict.keys()) == ['/var']
+        assert sorted(scan_dict['/var']['exclude']['items']) == ['/var/lib/misc', '/var/lib/systemd', '/var/log/wtmp']
+        assert all([x not in scan_dict['/var']['include']
+                    for x in ['/var/lib/misc', '/var/lib/systemd', '/var/log/wtmp', '/var/lib', '/var/log', '/tmp']])
+        assert all([x in scan_dict['/var']['include']
+                    for x in ['/var/cache', '/var/tmp', '/var/lib/xkb', '/var/lib/dbus',
+                              '/var/log/btmp', '/var/log/lastlog']])
+
+    @pytest.mark.skipif(not os.path.exists('/usr/local/libexec'), reason="No /usr/local/libexec")
+    def test_process_include_exclude_items_similar_names(self, conf, yara, rules, cmd):
```

Comment:
@mhuth this test fails on any computers that just happen to have files that match the pattern.  It would be better to create a tempdir that you know won't exist on a system and use that for your test.

```
>       assert len(usr_local_lib) > 2
E       AssertionError: assert 1 > 2
E        +  where 1 = len(['/usr/local/lib/libjpeg-x86_64.so.9.2.0'])

insights/tests/client/apps/test_malware_detection.py:928: AssertionError
```

Reply:
Ah yep.  Some (incorrect) assumptions going on there.  I'll fix that

Reply:
Ok, I removed that test.  Added another that uses tmp files instead.